packages/host/app/components/operator-mode/container.gts

```diff
@@ -601,14 +684,27 @@ export default class OperatorModeContainer extends Component<Signature> {
       <CardCatalogModal />
 
       <div class='operator-mode__with-chat {{this.chatVisibilityClass}}'>
-        <SubmodeSwitcher
-          @submode={{this.operatorModeStateService.state.submode}}
-          @onSubmodeSelect={{this.updateSubmode}}
-          class='submode-switcher'
-        />
+        <div class='operator-mode__top-menu'>
+          <SubmodeSwitcher
+            @submode={{this.operatorModeStateService.state.submode}}
+            @onSubmodeSelect={{this.updateSubmode}}
+          />
+          {{#if this.isCodeMode}}
+            <CardURLBar
+              @url={{this.codePath}}
+              @onEnterPressed={{this.updateCodePath}}
+              @loadFileError={{this.loadFileError}}
+              @resetLoadFileError={{this.resetLoadFileError}}
+              @realmInfo={{this.codeModeRealmInfo}}
+            />
```

Comment:
we have a component for code mode, why doesn't the URL bar live in that component?

Reply:
Because I need to make this component at the same level as submodel switcher button. And the submode switcher button is in the container not in the code mode.

Reply:
but why?

Reply:
Do you mean why the submode switcher in the container?

Reply:
no i mean why does having the submode switcher in the container mean that you can't put the URL bar in the code mode component?

Reply:
I did it to make style easier so the submode switcher and code bar URL can be in the same `div`.

Reply:
i think the CSS would be more sophisticated if you moved the URL bar into the code mode component, but that you could totally style it identically to how you have it now using absolute positioning. Personally I think the component hierarchy is more important than the CSS styling. The URL bar only ever appears in code mode so it seems very awkward to have it live outside of the component that is used for code mode.

Reply:
Also, I'm getting concerned about the size of the container.gts file. I think at some point soon we'll need to refactor it and split it apart. it feels like it's losing cohesion and doing to many jobs and thereby attracting other people to add ever more to it because it;'s not longer clear what its specific job is.  

@lukemelia might be a good idea to make a ticket to refactor the container.get component...

Reply:
Agree 

> Also, I'm getting concerned about the size of the container.gts file. I think at some point soon we'll need to refactor it and split it apart. it feels like it's losing cohesion and doing too many jobs and thereby attracting other people to add ever more to it because it's not longer clear what its specific job is.
> 
> @lukemelia might be a good idea to make a ticket to refactor the container.get component...

Agree, I think we can make sub folders for interact mode and code mode.

Reply:
> i think the CSS would be more sophisticated if you moved the URL bar into the code mode component, but that you could totally style it identically to how you have it now using absolute positioning. Personally I think the component hierarchy is more important than the CSS styling. The URL bar only ever appears in code mode so it seems very awkward to have it live outside of the component that is used for code mode.

Updated [152409f](https://github.com/cardstack/boxel/pull/571/commits/152409f06f02aae55163f488c1a8c30d2bb6fd03)